upup/pkg/fi/deletions.go

```diff
@@ -20,8 +20,12 @@ type ProducesDeletions[T SubContext] interface {
 	FindDeletions(*Context[T]) ([]Deletion[T], error)
 }
 
+type CloudupProducesDeletions = ProducesDeletions[CloudupSubContext]
```

Comment:
I don't think there will ever be a NodupProducesDeletions, so we could probably make this specific

Reply:
Yes, one of the things I like about your direction here is that we can clearly see this now (just a grep!)